### PR TITLE
Move thread parking to `sys::sync`

### PR DESCRIPTION
**Pull request by joboet**
Part of #117276.

I'll leave the platform-specific API abstractions in `sys::pal`, as per the initial proposal. I'm not entirely sure whether we'll want to keep it that way, but that remains to be seen.

r? @ChrisDenton (if you have time)

### CODE REVIEW

File: library/std/src/sys/sync/thread_parking/mod.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            target_os = "ios",
+            target_os = "watchos",
+            target_os = "visionos",
+            target_os = "tvos",
+        ),
+        not(miri),
+    ))] {
+        mod darwin;
+        pub use darwin::Parker;
+    } else if #[cfg(target_os = "netbsd")] {
```
[the remark this answers is not in the document]
Comment (joboet) [reply library/std/src/sys/sync/thread_parking/mod.rs:36]: No, it's not covered above.

[the remark this answers is not in the document]
Comment (joboet) [reply library/std/src/sys/sync/thread_parking/mod.rs:36]: Right, whoops!

File: library/std/src/sys/pal/unix/thread_parking.rs [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// Only used on NetBSD. If other platforms start using id-based parking, use
+// separate modules for each platform.
+#![cfg(target_os = "netbsd")]
```
Comment (ChrisDenton) [inline library/std/src/sys/pal/unix/thread_parking.rs:3]: Hmm... I'm not sure I love this being `unix/thread_parking.rs` with nothing in the path to indicate it's highly platform specific. If more unix targets do use id-based parking, what should they call their modules? How would that be organised?

Comment (joboet) [reply library/std/src/sys/pal/unix/thread_parking.rs:3]: Then I'd split them `unix/thread_parking/netbsd.rs`/`unix/thread_parking/*.rs` and expose the relevant inferface in `unix/thread_parking/mod.rs`.

**Comment by ChrisDenton** [comment]
Thanks

@bors r+

**Pull request merged**